### ISSUE TITLE
[BUG] Cannot use Scan or Import buttons on iOS

### ISSUE THREAD

**Issue by dirtymouse** [clawfire/covid19-passbook-generator#55]
On Apple iOS (v12, v13, v14) cannot action the scan or import button. Does nothing. A press is registered but nothing happens, therefor cannot create a EU C19 wallet

**Comment by clawfire** [comment]
Hi @dirtymouse 😃 

Can I ask which browser are you using? 
Sadly, accessing the camera require using the site in a plain browser and not an "in-app" browser like the one used in Messenger, Facebook, Twitter, ... That's a limitation both from the app creator (for Facebook, Messenger) and Apple (for Twitter, Instagram, ...).

Also, pushing a passbook in the Apple wallet requires using Safari only so far. 

We are working on some improvement to detect and tell users what to do when they are in that case #25

**Comment by dirtymouse** [comment]
I’m using stock safari browser in all cases

Everything is plain no blockers etc

Thanks - shane


————————————————
excuse my spelling or short reply
I'm using an iPhone...

Shane Wynter
Timezone: GMT
http://help.of.io/
iMessage/Phone/SMS: +353 89 9810201

**Comment by clawfire** [comment]
Do you have an ad blocker maybe? Like Hush ? I heard some user experiencing trouble with it and my app.

**Comment by dirtymouse** [comment]
All blockers disabled. I use blockerX but it’s fully disabled. Just checked it again. Same issue on three separate devices including one that has no blocker installed. Thanks 


————————————————
excuse my spelling or short reply
I'm using an iPhone...

Shane Wynter
Timezone: GMT
http://help.of.io/
iMessage/Phone/SMS: +353 89 9810201

**Comment by clawfire** [comment]
Could you try with the new update I pushed yesterday evening? 😃

**Comment by dirtymouse** [comment]
Hi Thibault ,

no luck, i tried again.

I can't believe there is no other way to get the EU QR code onto a wallet. Apple's wallet says it doesn't understand the data in the QR code if i try to add it manually. 

Thanks for looking into it - Shane



You previously wrote on Fri, 16 Jul 2021 00:46:21 -0700:
> 
> Could you try with the new update I pushed yesterday evening? 😃
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

-----------------------------
Shane Wynter

Timezone: GMT
http://help.of.io/
iMessage/Telegram/WAPP/Phone/SMS: +353 89 9810201

To send me a secure PGP encrypted message:
https://keybase.io/dirtymouse

**Comment by clawfire** [comment]
To be fair, Apple's wallet adding function by scanning a QR code is not really used outside the USA. It should be formated in a certain way and only for collecting coupons or getting a membership card, as I understood.

**Comment by clawfire** [comment]
Hey @dirtymouse , just wanted to check if you still have the issue with all the new stuff we released lately 😃

**Comment by dirtymouse** [comment]
Hi,


Thanks for checking in.

No change, clicking/touching the 'Import picture" or Start Scanning" doesn't do anything on iOS 12.5.3

on iOS 14.5, i get slightly further but 'Cannot download this file', see attached


Thanks - Shane

I



You previously wrote on Fri, 30 Jul 2021 18:29:48 -0700:
> 
> Hey @dirtymouse , just wanted to check if you still have the issue with all 
> the new stuff we released lately 😃
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

-----------------------------
Shane Wynter

Timezone: GMT
http://help.of.io/
iMessage/Telegram/WAPP/Phone/SMS: +353 89 9810201

To send me a secure PGP encrypted message:
https://keybase.io/dirtymouse

**Comment by dirtymouse** [comment]
Hey,

I had success doing this (below), and I've written instructions for your website

There are thousands of us struggling to get the QR code into Apple wallet. I've spoken with 5-6 people about this in the last week alone :)


>

ALTERNATIVE METHOD for iOS users (You can use a Mac or PC without a camera).

If you've received a PDF version of the QR Code, or taken a photo/scanned the QR code by other means:

[1] 
Crop the QR Code from the PDF/Scan/Photo and save as a JPG or PNG file to your Computer.

[2] 
Select 'Import a Picture' using Mac/PC with any Chromium based Web Brower

[3] 
Your QR code should load in the browser, verify the details are correct, click 'Save in your wallet'

[4] 
A file will download (to your downloads folder), called certificate.pkpass (allow if prompted)

[5] 
Send yourself an email or iMessage (that's also accessible on your iPhone) and attach the certificate.pkpass that's in your downloads folder

[6] 
View the message on your iPhone

- In iOS Mail (default app), you'll see a wallet icon as attachment, select it
- in iOS Messages: select the attachment

In both cases, when viewing the 'certificate.pkpass' attachment, press the blue 'Add' in the top right corner. 

[7]
This will save the EU Digital Covid Certification to your Wallet (app or lockscreen). There is no feedback to this setup. Go to your wallet to see if it was successful.

<









Perhaps these instructions might be added to your website, as this is beyond the realm of usability for 90% of people :)

Myself included, and I'm an IT guy with a github account using 2FA!


Thanks - Shane



You previously wrote on Fri, 30 Jul 2021 18:29:48 -0700:
> 
> Hey @dirtymouse , just wanted to check if you still have the issue with all 
> the new stuff we released lately 😃
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

-----------------------------
Shane Wynter

Timezone: GMT
http://help.of.io/
iMessage/Telegram/WAPP/Phone/SMS: +353 89 9810201

To send me a secure PGP encrypted message:
https://keybase.io/dirtymouse

**Comment by clawfire** [comment]
hey @dirtymouse 😃 
Thanks for this tuto. Indeed that's a way of doing it but it's very complicated. I wasn't sure I should integrate it to the site but ... probably will 😃. There's also a little shorter version if you're on a MacBook device 😃

**Comment by dirtymouse** [comment]
Hi,

Well, those were the hoops I had to jump through to get it to work.

If it worked on the iPhone, then that would be way better :)

Or am I the only one reporting problems?

You're probably missing out on loads of feedback as I'd say 99% of users don't have a github account to offer feedback to you :)

I see your site is being referenced online In Ireland, as one of the only ways to get QR codes into a wallet. 

Cheers - Shane

You previously wrote on Mon, 02 Aug 2021 13:01:18 -0700:
> 
> hey @dirtymouse 😃
> Thanks for this tuto. Indeed that's a way of doing it but it's very 
> complicated. I wasn't sure I should integrate it to the site but ... probably 
> will 😃. There's also a little shorter version if you're on a MacBook device 😃
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android.

-----------------------------
Shane Wynter

Timezone: GMT
http://help.of.io/
iMessage/Telegram/WAPP/Phone/SMS: +353 89 9810201

To send me a secure PGP encrypted message:
https://keybase.io/dirtymouse

**Comment by Wittner** [comment]
same problem as @dirtymouse I used my Linux desktop browser to click the button - scanned my paper copy QR - downloaded it and emailed the .pkpass to an email I have access to on my iPhone. Clicked on the attachment in the email on my iPhone and it recognised it as a Wallet item and I added it to Wallet from there. Not sure what' going on with your button. I'll have a look at your code and offer some help if I can. I think Safari on the iPhone has some weirdness around buttons anyhow.

**Comment by Wittner** [comment]
Have to leave my office and will be back in a couple of days. I cloned the repo to my local machine and had a quick look. Not sure exactly what stack you're using so I 'npm installed' and ran the 'start' script but I think I'm missing some local stuff (maybe parcel?). Anyhow I had a look at the code without running the app and noticed two things. You are using $('button[name="startScanning"]') to  add a click event but there are two buttons with the same name (for media queries?). I think that might cause an issue. Also you are using the name as an identifier in this instance but IDs in other instances. Perhaps using a different ID for each case. If they are both in the DOM at the same time with the same name and you're using 'name' as the selector, then that would be a red flag for me. You may already be aware of this but just thought I'd mention it.

**Comment by clawfire** [comment]
> You are using $('button[name="startScanning"]') to add a click event but there are two buttons with the same name (for media queries?)

hum ... I may not have pay attention enough when merging a PR 😅
I'll take a look at this

**Comment by clawfire** [comment]
Found out the bug was caused by the lack of support of `BigInt` in Safari before version 14. And there's no polyfill we can put in place to support this. `BigInt` is used to represent big integer numbers in memory, required for cryptographic manipulation such as decoding the QR code. 

I implemented a detection feature of lack of this functionality and warn the user with a message + prevent to discard the message (since the app will be totally unusable anyway).
